tests/integration/ha_tests/test_rollback_to_master_label.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import operator
+import shutil
+from pathlib import Path
+
+import pytest
+from pytest_operator.plugin import OpsTest
+from tenacity import Retrying, stop_after_attempt, wait_fixed
+
+from .. import markers
+from ..helpers import (
+    APPLICATION_NAME,
+    DATABASE_APP_NAME,
+    get_leader_unit,
+    get_primary,
+    get_unit_by_index,
+)
+from .helpers import (
+    are_writes_increasing,
+    check_writes,
+    get_instances_roles,
+    inject_dependency_fault,
+    start_continuous_writes,
+)
+
+logger = logging.getLogger(__name__)
+
+TIMEOUT = 600
+
+
+@pytest.mark.group(1)
+@markers.amd64_only  # TODO: remove after arm64 stable release
+@pytest.mark.abort_on_fail
+async def test_deploy_stable(ops_test: OpsTest) -> None:
+    """Simple test to ensure that the PostgreSQL and application charms get deployed."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            DATABASE_APP_NAME,
+            num_units=3,
+            channel="14/stable",
+            trust=True,
+        ),
+        ops_test.model.deploy(
+            APPLICATION_NAME,
+            num_units=1,
+            channel="latest/edge",
+        ),
+    )
+    logger.info("Wait for applications to become active")
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(
+            apps=[DATABASE_APP_NAME, APPLICATION_NAME], status="active"
```

Comment:
Optional of course, but if you want to avoid https://warthogs.atlassian.net/browse/DPE-4855 you can add `raise_on_error=False` in this specific wait.